platform/representation/src/gestures/DropGesture.js

```diff
@@ -40,14 +42,15 @@ define(
          * @param {DomainObject} domainObject the domain object whose
          *        composition should be modified as a result of the drop.
          */
-        function DropGesture(dndService, $q, element, domainObject) {
+        function DropGesture(dndService, $q, navigationService, objectService, instantiate, typeService, element, domainObject) {
             var actionCapability = domainObject.getCapability('action'),
+                editableDomainObject,
+                scope = element && element.scope && element.scope(),
```

Comment:
is element ever not an object?  Seems that constructing a drop gesture with an invalid element object is breaking the defined interface and should blow up, not silently fail.